testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs

```diff
@@ -847,26 +1096,68 @@ impl BitcoinRegtestController {
     fn finalize_tx(
         &mut self,
         tx: &mut Transaction,
-        budget_for_outputs: u64,
-        btc_miner_fee: u64,
-        mut utxos: Vec<UTXO>,
+        spent_in_outputs: u64,
+        spent_in_rbf: u64,
+        min_tx_size: u64,
+        fee_rate: u64,
+        utxos_set: &mut UTXOSet,
         signer: &mut BurnchainOpSigner,
-        attempt: u64,
     ) -> Option<()> {
         // spend UTXOs in decreasing order
-        utxos.sort_by(|u1, u2| u1.amount.cmp(&u2.amount));
-        utxos.reverse();
+        utxos_set.utxos.sort_by(|u1, u2| u1.amount.cmp(&u2.amount));
+        utxos_set.utxos.reverse();
+
+        let tx_size = {
+            // We will be calling 2 times serialize_tx, the first time with an estimated size,
+            // Second time with the actual size, computed thanks to the 1st attempt.
+            let estimated_rbf = if spent_in_rbf == 0 {
+                0
+            } else {
+                spent_in_rbf + min_tx_size // we're spending 1 sat / byte in RBF
+            };
+            let mut tx_cloned = tx.clone();
+            let mut utxos_cloned = utxos_set.clone();
+            self.serialize_tx(
+                &mut tx_cloned,
+                spent_in_outputs + min_tx_size * fee_rate + estimated_rbf,
+                &mut utxos_cloned,
+                signer,
+            );
+            let serialized_tx = SerializedTx::new(tx_cloned);
+            cmp::max(min_tx_size, serialized_tx.bytes.len() as u64)
+        };
+
+        let rbf_fee = if spent_in_rbf == 0 {
+            0
+        } else {
+            spent_in_rbf + tx_size // we're spending 1 sat / byte in RBF
+        };
+        self.serialize_tx(
+            tx,
+            spent_in_outputs + tx_size * fee_rate + rbf_fee,
+            utxos_set,
+            signer,
+        );
+        signer.dispose();
```

Comment:
Shouldn't this be called on a `impl Drop` for `BurnchainOpSigner`?  If not, can you make it so?